lib/Tool/Authentication.php

```diff
@@ -57,26 +68,89 @@ public static function authenticateSession(Request $request = null): ?User
             }
         }
 
-        if (!Session::requestHasSessionId($request, true)) {
-            // if no session cookie / ID no authentication possible, we don't need to start a session
+        $session = $request->hasPreviousSession() ? $request->getSession() : null;
+
+        if (null === $session) {
             return null;
         }
 
-        $session = Session::getReadOnly();
-        $user = $session->get('user');
+        $token = $session->get('_security_pimcore_admin');
+        $token = static::safelyUnserialize($token);
 
-        if ($user instanceof User) {
-            // renew user
-            $user = User::getById($user->getId());
+        if ($token instanceof TokenInterface) {
+            $token = static::refreshUser($token, \Pimcore::getContainer()->get(UserProvider::class));
+            $user = $token->getUser();
 
-            if (self::isValidUser($user)) {
-                return $user;
-            }
+            return $user instanceof \Pimcore\Bundle\AdminBundle\Security\User\User ? $user->getUser() : null;
```

Comment:
I can now login with inactive user credentials. Maybe we should still check a valid user here?

Reply:
changed

Reply:
Now `authenticateSession` returns nulls when user is inactive however i can still login. I guess we are are not validating submitted credentials anywhere in `login_path` or `check_path`.

Reply:
This is not resolved with just `authenticateSession` changes. We should update the `Adminbundle\Security\UserProvider::loadUserByIdentifier` with check:
```
$pimcoreUser = PimcoreUser::getByName($identifier);

        if ($pimcoreUser && $pimcoreUser->isActive()) {
```